Kiosk/Views/InkRecognizerExplorer/DualCanvas.xaml.cs

```diff
@@ -0,0 +1,860 @@
+        {
+            string recognizedText = recoUnit.recognizedText;
+
+            if (recognizedText != null)
+            {
+                int id = recoUnit.id;
+
+                // Color of ink word or ink bullet
+                var color = GetStrokeColor(recoUnit);
+
+                var text = new Tuple<string, Color>(recognizedText, color);
+                recoText.Add(id, text);
+            }
+        }
+
+        private void DrawText(InkRecognitionUnit recoUnit, CanvasControl sender, CanvasDrawEventArgs args)
+        {
+            var childIds = recoUnit.childIds;
+            var initialTransformation = args.DrawingSession.Transform;
+
+            // Points of bounding rectangle to align drawn text
+            float floatX = (float)recoUnit.boundingRectangle.topX;
+            float floatY = (float)recoUnit.boundingRectangle.topY;
+
+            // Rotated bounding rectangle points to get correct angle of text being drawn
+            float topLeftX = (float)recoUnit.rotatedBoundingRectangle[0].x;
+            float topLeftY = (float)recoUnit.rotatedBoundingRectangle[0].y;
+
+            float topRightX = (float)recoUnit.rotatedBoundingRectangle[1].x;
+            float topRightY = (float)recoUnit.rotatedBoundingRectangle[1].y;
+
+            float bottomRightX = (float)recoUnit.rotatedBoundingRectangle[2].x;
+            float bottomRightY = (float)recoUnit.rotatedBoundingRectangle[2].y;
+
+            float bottomLeftX = (float)recoUnit.rotatedBoundingRectangle[3].x;
+            float bottomLeftY = (float)recoUnit.rotatedBoundingRectangle[3].y;
+
+            // Height and width of bounding rectangle to get font size and width for text layout
+            float height = GetDistanceBetweenPoints(topRightX, bottomRightX, topRightY, bottomRightY) * dipsPerMm;
+            float width = GetDistanceBetweenPoints(bottomLeftX, bottomRightX, bottomLeftY, bottomRightY) * dipsPerMm;
+
+            if (height < 45)
+            {
+                height = 45;
+            }
+
+            // Transform to get correct angle of text
+            float centerPointX = ((topLeftX + topRightX) / 2) * dipsPerMm;
+            float centerPointY = ((topLeftY + bottomLeftY) / 2) * dipsPerMm;
+            var centerPoint = new Vector2(centerPointX, centerPointY);
+
+            Matrix3x2 angle = GetRotationAngle(bottomLeftX, bottomRightX, bottomLeftY, bottomRightY, centerPoint);
+
+            args.DrawingSession.Transform = angle;
+
+            var textFormat = new CanvasTextFormat()
+            {
+                FontSize = height - 5,
+                WordWrapping = CanvasWordWrapping.NoWrap,
+                FontFamily = "Ink Free"
+            };
+
+
+            // Build string to be drawn to canvas
+            string textLine = string.Empty;
+            foreach (var item in childIds)
+            {
+                int id = int.Parse(item.ToString());
+                (string text, _) = recoText[id];
```

Comment:
> (string text, _) [](start = 15, length = 17)

another construct I've not seen.  what does this do?

Reply:
Line 466 is deconstructing a tuple. I found it to be a little more readable than what I was doing previously since I was accessing the items of the tuple like recoText[id].Item1 and "Item1" is not very helpful when reading to code to understand what it is. Another option would be to give the values in the tuple an alias such as "Text" or "Color", but I found it pretty easy to read when it came down to code blocks like

`
            int index = 0;
            foreach (var item in childIds)
            {
                int id = int.Parse(item.ToString());
                (string text, Color color) = recoText[id];

                textLayout.SetColor(index, text.Length, color);

                index += text.Length + 1;
            }
`